### PR TITLE
Add filer command line parameter to let Filer UI show/hide directory delete button

### PR DESCRIPTION
**Pull request by blacktear23**
# What problem are we solving?

As #3149 mentioned, show delete directory button is danger. This PR will provide a filer command line parameter to let end user control the delete button show or not. 

# How are we solving the problem?

Provide `-ui.deleteDir` parameter for filer to control filer UI show or hide the delete button for directory. Default is true so if user want to disable delete directory button just add `-ui.deleteDir=false` parameter when start filer server.

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/command/filer.go [hunk 1]
```diff
@@ -82,6 +83,7 @@ func init() {
 	f.debug = cmdFiler.Flag.Bool("debug", false, "serves runtime profiling data, e.g., http://localhost:<debug.port>/debug/pprof/goroutine?debug=2")
 	f.debugPort = cmdFiler.Flag.Int("debug.port", 6060, "http port for debugging")
 	f.localSocket = cmdFiler.Flag.String("localSocket", "", "default to /tmp/seaweedfs-filer-<port>.sock")
+	f.showUIDirectoryDelete = cmdFiler.Flag.Bool("ui.deleteDir", false, "enable filer UI show delete directory button")
```
Comment (chrislusf) [inline weed/command/filer.go]: how about defaulting to true?

Comment (blacktear23) [reply weed/command/filer.go]: no problem. Do you think the parameter name change to `ui.disableDeleteDir` is acceptable?

Comment (chrislusf) [reply weed/command/filer.go]: "deleteDir" is ok.

Comment (blacktear23) [reply weed/command/filer.go]: So user should only using `ui.deleteDir=false` parameter to disable delete directory button. That seems strange. As normal usage, bool parameter always using false as default.

File: weed/command/filer.go [hunk 2]
```diff
@@ -82,6 +83,7 @@ func init() {
 	f.debug = cmdFiler.Flag.Bool("debug", false, "serves runtime profiling data, e.g., http://localhost:<debug.port>/debug/pprof/goroutine?debug=2")
 	f.debugPort = cmdFiler.Flag.Int("debug.port", 6060, "http port for debugging")
 	f.localSocket = cmdFiler.Flag.String("localSocket", "", "default to /tmp/seaweedfs-filer-<port>.sock")
+	f.showUIDirectoryDelete = cmdFiler.Flag.Bool("ui.deleteDir", true, "enable filer UI show delete directory button")
```
Comment (chrislusf) [inline weed/command/filer.go:86]: the server.go also need to add this option.

**Pull request merged**